app/logic/stake_logic.rb

```diff
@@ -68,15 +69,17 @@ def move_current_stakes_to_history
     end
   end
 
-  def save_stake_accounts
+  def update_stake_accounts
     lambda do |p|
       return p unless p.code == 200
 
       p.payload[:stake_accounts].each do |acc|
-        val = VoteAccount.find_by(
+        vote_account = VoteAccount.find_by(
           network: p.payload[:network],
           account: acc['delegatedVoteAccountAddress']
-        ).validator.id rescue nil
+        )
```

Comment:
```suggestion
        )&.validator_id
```
and then you can remove line below